salt/_pillar/metalk8s_nodes.py

```diff
@@ -50,13 +50,94 @@ def node_info(node, ca_minion):
     return result
 
 
+def cluster_version_info(namespace_information):
+    cluster_version = None
+
+    if namespace_information:
+        ns_info_dict = namespace_information.to_dict()
+        annotation_info = ns_info_dict.get(
+            'metadata', {}).get('annotations')
+        if annotation_info:
+            cluster_version = annotation_info.get(
+                'metalk8s.scality.com/cluster-version'
+            )
+    if not cluster_version:
+        cluster_version = __utils__['pillar_utils.errors_to_dict'](
+            'Unable to retrieve cluster version'
+        )
+
+    return cluster_version
+
+
+def get_storageclass(storageClassV1):
+    storage_classes = {}
+    storageclass_list = storageClassV1.list_storage_class()
+    if not storageclass_list:
+        return {}
+    for storageclass in storageclass_list.items:
+        storage_classes[storageclass.metadata.name] = storageclass
+
+    return storage_classes
+
+
+def volume_info(customObjectApi, minion_id, storageclassObject):
+    volume_information = {}
+    storage_classes = get_storageclass(storageclassObject)
+
+    try:
+        volume_response = customObjectApi.list_cluster_custom_object(
+            group="storage.metalk8s.scality.com", version="v1alpha1",
+            plural="volumes"
+        )
+
+        if not volume_response:
+            volume_information = __utils__['pillar_utils.errors_to_dict']([
+                'Unable to retrieve node volume list'
+            ])
+
+        for items in volume_response.get('items'):
+            if items['spec']['nodeName'] != minion_id:
+                continue
+            storageclass = storage_classes.get(
+                items['spec']['storageClassName']
+            )
+            if storageclass is None:
+                volume_information = __utils__['pillar_utils.errors_to_dict']([
+                    'No none storage class found for {}'.format(
+                        items['metadata']['name']
+                    )
+                ])
+            else:
+                result_vol_object = {
+                    'name': items['metadata']['name'],
+                    'uid': items['metadata']['uid'],
+                    'capacity': items.get('spec').get(
+                        'sparseLoopDevice', {}).get('size', ""),
+                    'storage_class': items.get('spec', {}).get(
+                        'storageClassName', ""),
+                    'status': items.get('status', {}).get('phase', ""),
+                    'storageclass_annotations':
+                        storageclass.metadata.annotations
+                }
+                volume_information[items['metadata']['name']] = \
+                    result_vol_object
+
+    except Exception as exc:
+        volume_information = __utils__['pillar_utils.errors_to_dict']([
```

Comment:
When faced with such errors, you could use an early `return` statement instead of setting the final return value in a variable, makes it much more readable (you know the function stops there).